mesonbuild/mesonmain.py

```diff
@@ -43,8 +44,8 @@ def errorhandler(e, command):
         if command == 'runpython':
             return 2
         elif isinstance(e, OSError):
-            mlog.exception("Unhandled python OSError. This is probably not a Meson bug, "
-                           "but an issue with your build environment.")
+            mlog.exception(Exception("Unhandled python OSError. This is probably not a Meson bug, "
```

Comment:
Shouldn't it be a `RuntimeError` instead?

Reply:
It's a bit of a code-smell to throw `Exception`, and it's only proper to use it as a base class, since the only way then to catch it is to catch `Exception`. `RuntimeError` is the sort of lowest generic exception that is usually thrown. I don't really care either way.

Reply:
But this isn't throwing anything...

Reply:
Good question.

It was introduced in https://github.com/mesonbuild/meson/commit/fa044e011d5d7a3525c9b3f2acfe1cbf4a0a96a1